src/components/time/time.spec.tsx

```diff
@@ -0,0 +1,1006 @@
+    expect(amToggle).toHaveAttribute("aria-pressed", "true");
+    expect(pmToggle).toBeInTheDocument();
+    expect(pmToggle).toHaveAttribute("aria-pressed", "false");
+  });
+
+  it("should display the AM/PM toggle and highlight the second button when toggleValue prop is `PM`", () => {
+    render(
+      <Time
+        value={{ hours: "", minutes: "", period: "PM" }}
+        onChange={() => {}}
+      />
+    );
+
+    const amToggle = screen.queryByText("AM");
+    const pmToggle = screen.queryByText("PM");
+
+    expect(amToggle).toBeInTheDocument();
+    expect(amToggle).toHaveAttribute("aria-pressed", "false");
+    expect(pmToggle).toBeInTheDocument();
+    expect(pmToggle).toHaveAttribute("aria-pressed", "true");
+  });
+
+  it("should render the input hint text when prop is set", () => {
+    render(
+      <Time
+        value={{ hours: "12", minutes: "30" }}
+        onChange={() => {}}
+        inputHint="hint text"
+      />
+    );
+
+    expect(screen.queryByText("hint text")).toBeInTheDocument();
+  });
+
+  it("should focus the relevant input when the associated label is clicked", async () => {
+    render(<Time value={{ hours: "12", minutes: "30" }} onChange={() => {}} />);
+
+    const user = userEvent.setup();
+    await user.click(screen.getByText("Hrs."));
+
+    expect(screen.getByDisplayValue("12")).toBeFocused();
+
+    await user.click(screen.getByText("Mins."));
+
+    expect(screen.getByDisplayValue("30")).toBeFocused();
+  });
+
+  it("should focus each input in the expected order when user is tabbing", async () => {
+    render(
+      <Time
+        value={{ hours: "12", minutes: "30", period: "AM" }}
+        onChange={() => {}}
+      />
+    );
+
+    const user = userEvent.setup();
+    await act(async () => {
+      await user.tab();
+    });
+
+    expect(screen.getByDisplayValue("12")).toBeFocused();
+
+    await act(async () => {
+      await user.tab();
+    });
+
+    expect(screen.getByDisplayValue("30")).toBeFocused();
+
+    await act(async () => {
+      await user.tab();
+    });
+
+    expect(screen.queryByText("AM")).toBeFocused();
+  });
+
+  it("should focus each input in the expected order when user is shift tabbing", async () => {
+    render(
+      <Time
+        value={{ hours: "12", minutes: "30", period: "AM" }}
+        onChange={() => {}}
+      />
+    );
+
+    const user = userEvent.setup();
+    const hrsInput = screen.getByDisplayValue("12");
+    const minsInput = screen.getByDisplayValue("30");
+    const amToggle = screen.queryByText("AM");
+
+    act(() => {
+      amToggle?.focus();
+    });
+
+    expect(amToggle).toHaveFocus();
+
+    await act(async () => {
+      await user.tab({ shift: true });
+    });
+
+    expect(minsInput).toBeFocused();
+
+    await act(async () => {
+      await user.tab({ shift: true });
+    });
+
+    expect(hrsInput).toBeFocused();
+  });
+
+  it("should render a legend with any passes label text", () => {
```

Comment:
question(non-blocking): is it meant to be this?

```suggestion
  it("should render a legend with any passed label text", () => {
```